### PR TITLE
Add dpdk perl-App-cpanminus hyperscan and dependencies to Mariner OS

### PR DESCRIPTION
**Pull request by chalamalasetty**
Add dpdk, perl-App-cpanminus, hyperscan and dependencies packages to Mariner OS

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Add dpdk, perl-App-cpanminus, hyperscan and dependencies to Mariner OS 

###### Change Log  

- Change
Add dpdk perl-App-cpanminus hyperscan and dependencies to Mariner OS

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Local build

### CODE REVIEW

File: SPECS/perl-Parse-PMFile/perl-Parse-PMFile.spec [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+BuildRequires:  perl(ExtUtils::MakeMaker) >= 6.76
+BuildRequires:  perl(strict)
+BuildRequires:  perl(warnings)
+# Run-time
+BuildRequires:  perl(Dumpvalue)
+BuildRequires:  perl(File::Spec)
+BuildRequires:  perl(JSON::PP) >= 2.00
+BuildRequires:  perl(Safe)
+BuildRequires:  perl(version) >= 0.83
+# Tests
```
Comment (thcrain-msft) [inline SPECS/perl-Parse-PMFile/perl-Parse-PMFile.spec:25]: If these are test dependencies, can we gate their inclusion at build-time behind the %with_check variable we set during check builds?

File: SPECS/perl-App-cpanminus/perl-App-cpanminus.spec [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+# Module::Signature is optional
+Requires:       perl-Parse-PMFile
+Requires:       perl(Safe)
+# version::vpp not used
+# Win32 not used
+Requires:       perl-YAML
+# XXX: Keep Provides: cpanminus to allow `yum install cpanminus' instead of
+# longer `yum install perl-App-cpanminus'.
+Provides:       cpanminus = %{version}-%{release}
+Obsoletes:      cpanminus <= 1.2002
```
Comment (thcrain-msft) [inline SPECS/perl-App-cpanminus/perl-App-cpanminus.spec:104]: This "Obsoletes" line is not necessary here, as we've never shipped a "cpanminus" package.

File: SPECS/perl-CPAN-DistnameInfo/perl-CPAN-DistnameInfo.spec [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+URL:            https://metacpan.org/release/CPAN-DistnameInfo
+Source0:        https://cpan.metacpan.org/authors/id/G/GB/GBARR/CPAN-DistnameInfo-%{version}.tar.gz
+BuildArch:      noarch
+BuildRequires:  make
+BuildRequires:  perl
+BuildRequires:  perl(version)
+BuildRequires:  perl(ExtUtils::MakeMaker) >= 6.76
+BuildRequires:  perl(strict)
+BuildRequires:  perl(warnings)
+# Tests:
```
Comment (thcrain-msft) [inline SPECS/perl-CPAN-DistnameInfo/perl-CPAN-DistnameInfo.spec:18]: If these are test dependencies, can we gate their inclusion at build-time behind the %with_check variable we set during check builds?

File: SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+BuildRequires:	perl(ExtUtils::MakeMaker) >= 6.76
+# Module
+BuildRequires:	perl(base)
+BuildRequires:	perl(CPAN::Meta::Prereqs) >= 2.132830
+BuildRequires:	perl(CPAN::Meta::Requirements) >= 2.121
+BuildRequires:	perl(Exporter)
+BuildRequires:	perl(Module::Metadata) >= 1.000023
+BuildRequires:	perl(strict)
+BuildRequires:	perl(warnings)
+# Test
```
Comment (thcrain-msft) [inline SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec:24]: If these are test dependencies, can we gate their inclusion at build-time behind the %with_check variable we set during check builds?

File: SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec [hunk 5]
```diff
@@ -0,0 +1,171 @@
+Name:		perl-CPAN-Meta-Check
+Summary:	Verify requirements in a CPAN::Meta object
+Version:	0.014
+Release:	12%{?dist}
+License:	GPL+ OR Artistic
```
Comment (thcrain-msft) [inline SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec:5]: >	 [](http://example.com/codeflow?start=8&length=1)

Nit: Can you align the tags in this section?

File: SPECS/perl-File-pushd/perl-File-pushd.spec [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+BuildRequires:  perl(Carp)
+BuildRequires:  perl(Cwd)
+BuildRequires:  perl(Exporter)
+BuildRequires:  perl(File::Path)
+BuildRequires:  perl(File::Spec)
+BuildRequires:  perl(File::Temp)
+BuildRequires:  perl(overload)
+BuildRequires:  perl(strict)
+BuildRequires:  perl(warnings)
+# Test Suite
```
Comment (thcrain-msft) [inline SPECS/perl-File-pushd/perl-File-pushd.spec:27]: If these are test dependencies, can we gate their inclusion at build-time behind the %with_check variable we set during check builds?

File: SPECS/perl-local-lib/perl-local-lib.spec [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+BuildRequires:  perl(File::Spec)
+BuildRequires:  perl(strict)
+BuildRequires:  perl(warnings)
+# Runtime
+BuildRequires:  perl(Carp)
+BuildRequires:  perl(Carp::Heavy)
+BuildRequires:  perl(Config)
+BuildRequires:  perl(File::Basename)
+BuildRequires:  perl(File::Glob)
+# Tests only
```
Comment (thcrain-msft) [inline SPECS/perl-local-lib/perl-local-lib.spec:31]: If these are test dependencies, can we gate their inclusion at build-time behind the %with_check variable we set during check builds?

File: SPECS/ragel/ragel.spec [hunk 8]
```diff
@@ -0,0 +1,231 @@
+Name:           ragel
+Version:        7.0.0.12
+Release:        4%{?dist}
+Summary:        Finite state machine compiler
+# aapl/ is the LGPLv2+
+License:        MIT AND LGPLv2+
+Group: 		Development/Libraries
```
Comment (thcrain-msft) [inline SPECS/ragel/ragel.spec:7]: >p: 		D [](http://example.com/codeflow?start=4&length=6)

Nit: this tag is misaligned

File: SPECS/dpdk/dpdk.spec [hunk 9]
```diff
@@ -0,0 +1,523 @@
...
+Example applications utilizing the Data Plane Development Kit, such
+as L2 and L3 forwarding.
+%endif
+
+%define sdkdir  %{_datadir}/%{name}
+%define docdir  %{_docdir}/%{name}
+%define incdir %{_includedir}/%{name}
+%define pmddir %{_libdir}/%{name}-pmds
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
```
Comment (thcrain-msft) [inline SPECS/dpdk/dpdk.spec:106]: Can this be moved to the top section, where we usually have these tags?

File: SPECS/perl-Parse-PMFile/perl-Parse-PMFile.spec (hunk 1 above)
Comment (chalamalasetty) [reply SPECS/perl-Parse-PMFile/perl-Parse-PMFile.spec:25]: Added %with_check for tests.

File: SPECS/perl-CPAN-DistnameInfo/perl-CPAN-DistnameInfo.spec (hunk 3 above)
Comment (chalamalasetty) [reply SPECS/perl-CPAN-DistnameInfo/perl-CPAN-DistnameInfo.spec:18]: added %with_check for tests

File: SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec (hunk 5 above)
Comment (chalamalasetty) [reply SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec]: Fixed alignment.

File: SPECS/perl-File-pushd/perl-File-pushd.spec (hunk 6 above)
Comment (chalamalasetty) [reply SPECS/perl-File-pushd/perl-File-pushd.spec:27]: Added %with_check for tests

File: SPECS/perl-local-lib/perl-local-lib.spec (hunk 7 above)
Comment (chalamalasetty) [reply SPECS/perl-local-lib/perl-local-lib.spec:31]: added %with_check for tests

File: SPECS/ragel/ragel.spec (hunk 8 above)
Comment (chalamalasetty) [reply SPECS/ragel/ragel.spec]: fixed aligment

File: SPECS/dpdk/dpdk.spec (hunk 9 above)
Comment (chalamalasetty) [reply SPECS/dpdk/dpdk.spec]: Removed duplicate.

File: SPECS/perl-App-cpanminus/perl-App-cpanminus.spec (hunk 2 above)
Comment (chalamalasetty) [reply SPECS/perl-App-cpanminus/perl-App-cpanminus.spec]: Removed.

File: SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec (hunk 4 above)
Comment (chalamalasetty) [reply SPECS/perl-CPAN-Meta-Check/perl-CPAN-Meta-Check.spec:24]: Added with_check for tests

**Review by thcrain-msft — APPROVED** [review]

**Review by neha170 — APPROVED** [review]

**Pull request merged**